sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -767,6 +771,10 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
           if useAdvanced =>
         Some(convertInToOr(name, values))
 
+      case Not(In(ExtractAttribute(SupportedAttribute(name)), ExtractableLiterals(values)))
```

Comment:
`Not(a IN (null, 2))`: if a = 1, the final result is null.
`a != 2`: if a = 1, the final result is true

I think this rewrite is incorrect. We need to make sure the values of IN are all not null.

Reply:
My bad, missed this. For `Not(InSet)` case, null value can not be skip safely, it should convert to `and(xx = null)`.